Core/test/qa-functional/src/org/sleuthkit/autopsy/ingest/EmbeddedFileTest.java

```diff
@@ -0,0 +1,194 @@
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        } 
+    }
+
+    @Override
+    public void tearDown() {
+        try {
+            Case.closeCurrentCase();
+            //Seems like we need some time to close the case.
+            try {
+                Thread.sleep(2000);
+            } catch (Exception ex) {
+
+            }
+
+            FileUtils.deleteDirectory(CASE_DIR);
+        } catch (CaseActionException | IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        assertFalse(CASE_DIR.exists());
+    }
+    
+    public void testEncription() {
+        try {
+            List<AbstractFile> results = openCase.getSleuthkitCase().findAllFilesWhere("name LIKE '%%'");            
+            String protectedName1 = "password_protected.zip";
+            String protectedName2 = "level1_protected.zip";
+            String protectedName3 =  "42.zip";
+            assertEquals(2207, results.size());
+            int passwdProtectedZips = 0;
+            for (AbstractFile file : results) {
+                //.zip file has artifact TSK_ENCRYPTION_DETECTED
+                if (file.getName().equalsIgnoreCase(protectedName1) || file.getName().equalsIgnoreCase(protectedName2) || file.getName().equalsIgnoreCase(protectedName3)){
+                    ArrayList<BlackboardArtifact> artifacts = file.getAllArtifacts();
+                    assertEquals(1, artifacts.size());
+                    for (BlackboardArtifact artifact : artifacts) {
+                        assertEquals(artifact.getArtifactTypeID(), BlackboardArtifact.ARTIFACT_TYPE.TSK_ENCRYPTION_DETECTED.getTypeID());
+                        passwdProtectedZips++;
+                    }
+                } else {//No other files has artifact defined
+                    assertEquals(0, file.getAllArtifacts().size());
+                }
+                
+                
+            }
+            //Make sure 3 password protected zip files has been tested: password_protected.zip, level1_protected.zip and 42.zip that we download for bomb testing.
+            assertEquals(3, passwdProtectedZips);
+        } catch (TskCoreException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        }
+        
+    }
+
+    private void runIngestJob(List<Content> datasources) {
+        IngestModuleTemplate embeddedTemplate = getIngestModuleTemplate(new EmbeddedFileExtractorModuleFactory());
+        IngestModuleTemplate hashLookupTemplate = getIngestModuleTemplate(new HashLookupModuleFactory());
+
+        ArrayList<IngestModuleTemplate> templates = new ArrayList<>();
+        templates.add(embeddedTemplate);
+        templates.add(hashLookupTemplate);
+        IngestJobSettings ingestJobSettings = new IngestJobSettings(IngestFileFiltersTest.class.getCanonicalName(), IngestType.FILES_ONLY, templates);
+        try {
```

Comment:
It looks like the contents of the try block and the getIngestModuleTemplate method are good candidates for utility methods to be shared across tests.